<launch>
     <node pkg="imu_driver" type="imu_node" name="imu_node" output="screen" >
		<param name="baudrate" type="int" value="115200" />
		<param name="port" type="string" value="/dev/ttyUSB0" />
		<param name="looprate" type="int" value="50" />
		<remap from="imu" to="imu/data_raw" />
     </node>

    <!-- Filter and fuse raw imu data -->
    <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_madgwick" output="screen" respawn="false" >
        <param name="fixed_frame" value="base_footprint" />
        <param name="use_mag" value="false" />
        <param name="publish_tf" value="false" />
        <param name="use_magnetic_field_msg" value="true" /> 
        <param name="world_frame" value="enu" />
        <param name="orientation_stddev" value="0.05" />
    </node>

        
    <!-- Publish static transform from base_footprint to imu_link -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_footprint_to_imu_link" args="0 0 0 0 0 0  /base_footprint /imu_link"/>
</launch>
